sisyphus/manager.py

```diff
@@ -572,6 +576,12 @@ def maybe_clear_state(state, always_clear, action):
             self.job_cleaner.start()
         return True
 
+    def handle_job_failure(self, prev_jobs: Dict[str, List[Job]], cur_jobs: Dict[str, List[Job]]):
+        prev_errored_jobs = set(prev_jobs.get(gs.STATE_ERROR, []))
+        for job in cur_jobs.get(gs.STATE_ERROR, []):
+            if job not in prev_errored_jobs:
```

Comment:
do we need to keep track of which job is newly added to error state or could this be a stateless function and we always use it on all errored jobs?